corehq/apps/locations/dbaccessors.py

```diff
@@ -1,31 +1,42 @@
-from corehq.apps.users.models import CommCareUser
-
-
-def _users_by_location(location_id, include_docs=True, wrap=True):
+def _cc_users_by_location(location_id, include_docs=True, wrap=True):
+    from corehq.apps.users.models import CommCareUser
     view = CommCareUser.view if wrap else CommCareUser.get_db().view
     return view(
         'locations/users_by_location_id',
-        startkey=[location_id],
-        endkey=[location_id, {}],
+        startkey=[location_id, CommCareUser.__name__],
+        endkey=[location_id, CommCareUser.__name__, {}],
         include_docs=include_docs,
+        reduce=False,
     )
 
 
 def get_users_by_location_id(location_id):
     """
-    Get all users for a given location
+    Get all mobile_workers for a given location
     """
-    return _users_by_location(location_id).all()
+    return _cc_users_by_location(location_id)
 
 
 def get_user_docs_by_location(location_id):
-    return _users_by_location(location_id, wrap=False).all()
+    return _cc_users_by_location(location_id, wrap=False).all()
 
 
 def get_user_ids_by_location(location_id):
     return [user['id'] for user in
-            _users_by_location(location_id, include_docs=False, wrap=False)]
+            _cc_users_by_location(location_id, include_docs=False)]
 
 
 def get_one_user_at_location(location_id):
-    return _users_by_location(location_id).first()
+    return _cc_users_by_location(location_id).first()
+
+
+def get_all_users_by_location(location_id):
+    from corehq.apps.users.models import CouchUser
+    results = CouchUser.get_db().view(
+        'locations/users_by_location_id',
+        startkey=[location_id],
+        endkey=[location_id, {}],
```

Comment:
The original view didn't have it, but I agree that it's probably a good check to enforce.  I can't really come up with a good scenario where you'd use this, but wouldn't have access to the domain.